infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -19,84 +19,106 @@
     private long recvSeq;
     private long shadowStreamStartAddress;
+
     @Getter
     private LogReplicationMetadataManager logReplicationMetadataManager;
-    HashMap<UUID, UUID> uuidMap;
-    Phase phase;
-
-
-    // The sequence number of the message, it has received.
-    // It is expecting the message in order of the sequence.
+    private HashMap<UUID, UUID> uuidMap;
+    private Phase phase;
 
     public StreamsSnapshotWriter(CorfuRuntime rt, LogReplicationConfig config, LogReplicationMetadataManager logReplicationMetadataManager) {
         this.rt = rt;
         this.logReplicationMetadataManager = logReplicationMetadataManager;
-        streamViewMap = new HashMap<>();
-        uuidMap = new HashMap<>();
-        shadowMap = new HashMap<>();
-        phase = Phase.TransferPhase;
-
-        for (String stream : config.getStreamsToReplicate()) {
-            String shadowStream = stream + SHADOW_STREAM_NAME_SUFFIX;
-            UUID streamID = CorfuRuntime.getStreamID(stream);
-            UUID shadowID = CorfuRuntime.getStreamID(shadowStream);
-            uuidMap.put(streamID, shadowID);
-            uuidMap.put(shadowID, streamID);
-            streamViewMap.put(streamID, stream);
-            shadowMap.put(shadowID, shadowStream);
+        this.streamViewMap = new HashMap<>();
+        this.uuidMap = new HashMap<>();
+        this.phase = Phase.TRANSFER_PHASE;
+
+        initializeShadowStreams(config);
+    }
+
+    /**
+     * Create shadow streams.
+     *
+     * We create a shadow stream per stream to replicate. A shadow stream aims to accumulate updates
+     * temporarily while the (full) snapshot sync completes. Shadow streams aim to avoid inconsistent
+     * states while data is still being transferred from active to standby.
+     *
+     * We currently, wait for snapshot sync to complete before applying data in shadow streams
+     * to the actual streams, this means that there is still a window of inconsistency as apply is not atomic,
+     * but at least inconsistency is at a point where there is guarantee of all data being available on the receiver.
+     * In the future, we will support Table Aliasing which will enable atomic flip from shadow to regular streams, avoiding
+     * complete inconsistency.
+     */
+    private void initializeShadowStreams(LogReplicationConfig config) {
+        // For every stream create a shadow stream which name is unique based
+        // on the original stream and a suffix.
+        for (String streamName : config.getStreamsToReplicate()) {
+            String shadowStreamName = streamName + SHADOW_STREAM_SUFFIX;
+            UUID streamId = CorfuRuntime.getStreamID(streamName);
+            UUID shadowStreamId = CorfuRuntime.getStreamID(shadowStreamName);
+            uuidMap.put(streamId, shadowStreamId);
+            uuidMap.put(shadowStreamId, streamId);
+            streamViewMap.put(streamId, streamName);
+
```

Comment:
streamViewMap should not change after initialized?

Reply:
done.